### PR TITLE
fix: reduce code issues

### PR DESCRIPTION
**Pull request by viceice**
## Changes:
Fix some codeql isues:
- `gradle-lite`: greedy regex

## Context:

Fixes CodeQL errors

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by monstro805 — DISMISSED** [review]

File: lib/manager/dockerfile/__fixtures__/2.Dockerfile [hunk 1]
```diff
@@ -21,23 +21,13 @@ FROM\
...
-
-COPY\
-    --chown=root \
-	# comment1
-#
- #comment2
- --from=image10
-	# comment2
-    path1 path2
-
```
Comment (viceice) [inline lib/manager/dockerfile/__fixtures__/2.Dockerfile:40]: Invalid docker syntax

File: lib/manager/dockerfile/__snapshots__/extract.spec.ts.snap [hunk 2]
```diff
@@ -89,24 +89,6 @@ Array [
...
-  },
-  Object {
-    "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
-    "currentDigest": undefined,
-    "currentValue": undefined,
-    "datasource": "docker",
-    "depName": "image10",
-    "depType": "stage",
-    "replaceString": "image10",
-  },
```
Comment (viceice) [inline lib/manager/dockerfile/__snapshots__/extract.spec.ts.snap:109]: From invalid docker syntax

File: lib/manager/dockerfile/extract.spec.ts [hunk 3]
```diff
@@ -2,12 +2,12 @@ import { readFileSync } from 'fs';
...
 
 const d1 = readFileSync(
-  'lib/manager/dockerfile/__fixtures__/Dockerfile1',
+  'lib/manager/dockerfile/__fixtures__/1.Dockerfile',
   'utf8'
 );
 
 const d2 = readFileSync(
-  'lib/manager/dockerfile/__fixtures__/Dockerfile2',
+  'lib/manager/dockerfile/__fixtures__/2.Dockerfile',
```
Comment (viceice) [inline lib/manager/dockerfile/extract.spec.ts:10]: For syntax highlighting in vscode 😉

File: lib/manager/gradle-lite/utils.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ import {
 } from './common';
 
 const artifactRegex = regEx(
-  '^[a-zA-Z][-_a-zA-Z0-9]*(?:.[a-zA-Z0-9][-_a-zA-Z0-9]*)*$'
+  '^[a-zA-Z][-_a-zA-Z0-9]*(?:\\.[a-zA-Z0-9][-_a-zA-Z0-9]*?)*$'
```
Comment (viceice) [inline lib/manager/gradle-lite/utils.ts:13]: only match dot

**Review by rarkins — APPROVED** [review]

**Pull request merged**